### PR TITLE
Building for MIC

### PR DESCRIPTION
**Pull request by zzlei**
There're some minor modifications to autoconf scripts and a source file.
A tutorial is also added, named README-MIC.

### CODE REVIEW

File: doc/README-MIC [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    OpenSSL (libssl & libcrypto)
+
+They can be downloaded from their websites. But building them require some 
+effort, and only the mentioned versions are guaranteed to work.  
+Assuming those libraries are to be install under path $MIC and the path to
+JtR is $JOHN, then follow the steps below.
+
+Build Zlib (version 1.2.8):
+$ cd */zlib-1.2.8
+$ CC=icc CFLAGS="-mmic" ./configure --prefix=$MIC
```
Comment (magnumripper) [inline doc/README-MIC]: The normal way to pass things to configure is on the right side, as in
```
./configure CC=icc CFLAGS="-mmic" --prefix=$MIC
```
Even if having them on the left side "works" there are subtle differences that should be avoided.

**Comment by magnumripper** [comment]
Looks good to me. I think putting those external patches in unused/ was the right thing for now. Please correct the `./configure` syntax and I'll merge this.

BTW we have never had to patch config.sub before. The one we had was verbatim from autoconf. Did you do any research leading up to this solution? I'm not saying you did wrong at all, I'm just not sure.

**Comment by frank-dittrich** [comment]
> BTW we have never had to patch config.sub before. The one we had was verbatim from autoconf. Did you do any research leading up to this solution? I'm not saying you did wrong at all, I'm just not sure.

I don't think it is wrong. If you look at the patches in unused, you'll see the same change for those libraries autoconf.
I guess it is just lacking mic support in the particular autoconf versions.
May be something to report upstream?

**Comment by magnumripper** [comment]
> May be something to report upstream?

That's part of what I meant. Maybe there *is* a newer version available upstream.

**Comment by jfoug** [comment]
NOTE, we are 9 months to a year out of date on the 'core' autoconf stuff.  mic*) may already be in this file, we just have not check for updates ourselves.

**Comment by zzlei** [comment]
I checked out autoconf's latest source code. It still doesn't recognise MIC. Actually config.sub was last updated in 2013.
We may consider submitting a patch to their community.

File: doc/README-MIC (hunk 1 above)
Comment (zzlei) [inline doc/README-MIC]: That's fixed now. Thank you.
BTW, I'm still working on removing the "-lssl -lcrypto" flags from LDFLAGS when building JtR for MIC.

**Pull request merged**